modules/core/src/object.cpp

```diff
@@ -1,32 +1,64 @@
-/*
-    This file is part of WARG's computer-vision
-
-    Copyright (c) 2015, Waterloo Aerial Robotics Group (WARG)
-    All rights reserved.
-
-    Redistribution and use in source and binary forms, with or without
-    modification, are permitted provided that the following conditions are met:
-    1. Redistributions of source code must retain the above copyright
-       notice, this list of conditions and the following disclaimer.
-    2. Redistributions in binary form must reproduce the above copyright
-       notice, this list of conditions and the following disclaimer in the
-       documentation and/or other materials provided with the distribution.
-    3. Usage of this code MUST be explicitly referenced to WARG and this code
-       cannot be used in any competition against WARG.
-    4. Neither the name of the WARG nor the names of its contributors may be used
-       to endorse or promote products derived from this software without specific
-       prior written permission.
-
-    THIS SOFTWARE IS PROVIDED BY WARG ''AS IS'' AND ANY
-    EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
-    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-    DISCLAIMED. IN NO EVENT SHALL WARG BE LIABLE FOR ANY
-    DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
-    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
-    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
-    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-*/
+/**
+ * @file object.cpp
+ * @author WARG
+ *
+ * @section LICENSE
+ *
+ *  Copyright (c) 2015-2017, Waterloo Aerial Robotics Group (WARG)
+ *  All rights reserved.
+ *
+ *  This software is licensed under a modified version of the BSD 3 clause license
+ *  that should have been included with this software in a file called COPYING.txt
+ *  Otherwise it is available at:
+ *  https://raw.githubusercontent.com/UWARG/computer-vision/master/COPYING.txt
+ */
+
+
+#include "object.h"
+#include "pixel_object.h"
+
+Object::Object(){
+
+}
+
+void Object::add_pobject(PixelObject * po){
+    pixelObjects.push_back(po);
+}
+
+cv::Point2d Object::get_centroid(){
+    return centroid;
+}
+
+cv::Scalar Object::get_colour(){
+    return colour;
+}
+
+void Object::update(){
+    int n = Object::pixelObjects.size();
+    this->centroid = cv::Point2d(0,0);
+    cv::Point2d maxDistance(0,0);
+    cv::Scalar cSum;
+    for (PixelObject* po : pixelObjects){
+        this->centroid += po->get_gps_centroid();
+        this->area += po->get_gps_area();
```

Comment:
This entire file isn't being used....I'll leave it in case, follow up work needs to be done for next competition.